compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.td

```diff
@@ -142,6 +142,30 @@ def EncodingAttr :
   let genVerifyDecl = 1;
 }
 
+//===---------------------------------------------------------------------===//
+// encoding.matmul_k
+//===---------------------------------------------------------------------===//
+
+def MatmulKAttr : IREEEncoding_Attr<"MatmulK"> {
+  let mnemonic = "matmul_k";
+  let summary = [{An attribute that tracks reduction dimensions for matmul}];
+  let description = [{
+    The attribute is specialized for tracking the matmul reduction dimensions
+    only. The encoded dimensions are the reduction dimensions that will be
+    further used in a matmul.
```

Comment:
Can you explain what is being tracked (dim sizes) and what happens in the dynamic case?

Reply:
It does not track the dimension sizes. Instead, it tracks what dimensions are reduction dimensions. E.g., the lhs tensor has `k_dims = [1]` when the indexing map is `(M, N, K) -> (M, K)`. It becomes `k_dims = [2]` if the indexing map is `(B, M, N, K) -> (B, M, K)`.

In this context, there is no difference between static case and dynamic case. Users can query the size from the tensor type. This is also why transforming on reduction dimensions will lead to an undefined behavior.

Reply:
Can you add a similar explanation to the attr description? This wasn't immediately obvious to me.

Reply:
Sure, will do!